be/test/storage/persistent_index_test.cpp

```diff
@@ -89,4 +91,35 @@ PARALLEL_TEST(PersistentIndexTest, test_mutable_index) {
     ASSERT_EQ(upsert_not_found.key_idxes.size(), expect_not_found);
 }
 
+PARALLEL_TEST(PersistentIndexTest, test_mutable_flush_to_immutable) {
+    using Key = uint64_t;
+    int N = 200000;
+    vector<Key> keys(N);
+    vector<IndexValue> values(N);
+    for (int i = 0; i < N; i++) {
+        char* dst = (char*)&keys[i];
+        snprintf(dst, 8, "%07d", i);
+        values[i] = i * 2;
+    }
+    auto rs = MutableIndex::create(sizeof(Key));
+    ASSERT_TRUE(rs.ok());
+    std::unique_ptr<MutableIndex> idx = std::move(rs).value();
+
+    // test insert
+    ASSERT_TRUE(idx->insert(keys.size(), keys.data(), values.data()).ok());
+
+    //auto env = std::make_unique<EnvMemory>();
+    auto env = Env::Default();
+    auto block_mgr = std::make_unique<fs::FileBlockManager>(env, fs::BlockManagerOptions());
+    std::unique_ptr<fs::WritableBlock> wblock;
+    fs::CreateBlockOptions opts({"./index.l1.1.1"});
+    ASSERT_TRUE(block_mgr->create_block(opts, &wblock).ok());
+    auto st = idx->flush_to_immutable_index(idx->size(), EditVersion(1, 1), *wblock);
+    if (!st.ok()) {
+        LOG(WARNING) << st;
+    }
+    ASSERT_TRUE(wblock->finalize().ok());
```

Comment:
Is there necessary to load the immutable index to check the write's success?

Reply:
This will be in following PR

Reply:
fixed